Sources/Thread.swift

```diff
@@ -65,16 +65,28 @@ public final class Thread {
 
 // MARK: - Private
 
-private func ThreadPrivateMain(arg: UnsafeMutableRawPointer?) -> UnsafeMutableRawPointer? {
-    
-    let unmanaged = Unmanaged<Thread.Closure>.fromOpaque(arg!)
+// This double declaration is needed becaus of different `pthread_create` parameters requirements, between osx and linux (arm)
+#if os(OSX) || os(iOS) || os(watchOS) || os(tvOS)
+private func ThreadPrivateMain(arg: UnsafeMutableRawPointer) -> UnsafeMutableRawPointer? {
+    let unmanaged = Unmanaged<Thread.Closure>.fromOpaque(arg)
     
     unmanaged.takeUnretainedValue().closure()
     
     unmanaged.release()
     
     return nil
 }
+#elseif os(Linux)
+private func ThreadPrivateMain(arg: UnsafeMutableRawPointer?) -> UnsafeMutableRawPointer? {
```

Comment:
We can just have the first two lines of the function between #if os(OSX),#elseif,#endif, the rest , from unmanaged.takeUnretainedValue() onwards can remain outside of the conditional block.